tests/unit/expire.tcl

```diff
@@ -601,4 +601,117 @@ start_server {tags {"expire"}} {
            {del foo}
         }
     } {} {needs:repl}
+
+    test {EXPIRE with NX option on a key with ttl} {
+        r SET foo bar EX 100000
+        assert_equal [r EXPIRE foo 200000 NX] 0
+        assert_range [r TTL foo] 50000 100000
+    } {}
+
+    test {EXPIRE with NX option on a key without ttl} {
+        r SET foo bar
+        assert_equal [r EXPIRE foo 200000 NX] 1
+        assert_range [r TTL foo] 100000 200000
+    } {}
+
+    test {PEXPIRE with NX option on a key with ttl} {
+        r SET foo bar EX 100000
+        assert_equal [r PEXPIRE foo 200000 NX] 0
+        assert_range [r TTL foo] 50000 100000
+    } {}
+
+    test {PEXPIRE with NX option on a key without ttl} {
+        r SET foo bar
+        assert_equal [r PEXPIRE foo 200000 NX] 1
+        assert_range [r TTL foo] 100 200
+    } {}
+
+    test {EXPIRE with XX option on a key with ttl} {
+        r SET foo bar EX 100000
+        assert_equal [r EXPIRE foo 200000 XX] 1
+        assert_range [r TTL foo] 100000 200000
+    } {}
+
+    test {EXPIRE with XX option on a key without ttl} {
+        r SET foo bar
+        assert_equal [r EXPIRE foo 200000 XX] 0
+        assert_equal [r TTL foo] -1
+    } {}
+
+    test {EXPIRE with GT option} {
+        r SET foo bar EX 100000
+        assert_equal [r EXPIRE foo 200000 GT] 1
+        assert_range [r TTL foo] 100000 200000
+    } {}
+
+    test {EXPIRE with GT option} {
```

Comment:
the previous test has the same name, let's rename one of them, or both.
i.e. the first one is `EXPIRE with GT on a key with low ttl`
the second one is `EXPIRE with GT on a key with high ttl`
and we do have a test below that's `EXPIRE with GT option on a key without ttl`